app/questionnaire/questionnaire_schema.py

```diff
@@ -268,19 +268,26 @@ def _update_answer_dependencies_for_calculated_summary(
             }
 
     def _update_answer_dependencies_for_calculations(
-        self,
-        calculations: tuple[ImmutableDict[str, Any]],
+        self, calculations: tuple[ImmutableDict[str, Any]], block_id: str
     ) -> None:
         for calculation in calculations:
             if not (source_answer_id := calculation.get("answer_id")):
-                continue
+                if not (value := calculation.get("value")):
+                    continue
+                if isinstance(value, dict):
+                    self._update_answer_dependencies_for_value_source(
+                        value,
+                        block_id=block_id,
+                    )
+                    continue
+
             dependents = {
                 self._get_answer_dependent_for_block_id(
                     block_id=self.get_block_for_answer_id(answer_id)["id"]  # type: ignore
                 )
                 for answer_id in calculation["answers_to_calculate"]
             }
-            self._answer_dependencies_map[source_answer_id] |= dependents
+            self._answer_dependencies_map[source_answer_id] |= dependents  # type: ignore
```

Comment:
This shouldn't be type ignored. The logic allows it to get this far without `source_answer_id`.

I would update to something like:
```py
            if source_answer_id := calculation.get("answer_id"):
                dependents = {
                    self._get_answer_dependent_for_block_id(
                        block_id=self.get_block_for_answer_id(answer_id)["id"]  # type: ignore
                    )
                    for answer_id in calculation["answers_to_calculate"]
                }
                self._answer_dependencies_map[source_answer_id] |= dependents

            elif isinstance(calculation.get("value"), dict):
                self._update_answer_dependencies_for_value_source(
                    calculation["value"],
                    block_id=block_id,
                )
```

*Note: not tested with suggested change.*

Reply:
Ok, changed it to your logic now, works fine.